### PR TITLE
feat: enable environment settings on Docker compose files

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

This PR modifies the Docker compose files to enable the environment reported on to the APM Server. Also, we've added editorconfig settings for YAML files.

## Why is it important?

This allows see the environment in the APM UI.

### CODE REVIEW

File: docker-compose-elastic-cloud.yml [hunk 1]
```diff
@@ -23,6 +23,11 @@ services:
       - ELASTIC_APM_SOURCE_LINES_SPAN_APP_FRAMES=5
       - ELASTIC_APM_SOURCE_LINES_ERROR_LIBRARY_FRAMES
       - ELASTIC_APM_SOURCE_LINES_SPAN_LIBRARY_FRAMES
+      - ELASTIC_APM_ENVIRONMENT=production
+      - ELASTIC_APM_ENABLE_LOG_CORRELATION=true
+      - ELASTIC_APM_LOG_LEVEL=DEBUG
+      - ENABLE_JSON_LOGGING=true
+      - ELASTIC_APM_USE_STRUCTLOG=true
```
Comment (beniwohli) [inline docker-compose-elastic-cloud.yml]: FYI, Both of these are already set to `True` in the Dockerfile

File: docker-compose-elastic-cloud.yml [hunk 2]
```diff
@@ -23,6 +23,11 @@ services:
       - ELASTIC_APM_SOURCE_LINES_SPAN_APP_FRAMES=5
       - ELASTIC_APM_SOURCE_LINES_ERROR_LIBRARY_FRAMES
       - ELASTIC_APM_SOURCE_LINES_SPAN_LIBRARY_FRAMES
+      - ELASTIC_APM_ENVIRONMENT=production
+      - ELASTIC_APM_ENABLE_LOG_CORRELATION=true
```
Comment (beniwohli) [inline docker-compose-elastic-cloud.yml]: I haven't seen this setting before. It's not a flag we have in the Python agent. Do other agents have this flag or where does it come from?

Comment (kuisathaverat) [reply docker-compose-elastic-cloud.yml]: 😞 it is only for the Java Agent https://www.elastic.co/guide/en/apm/agent/java/current/config-logging.html#config-enable-log-correlation

File: .editorconfig [hunk 3]
```diff
@@ -11,3 +11,7 @@ insert_final_newline = true
 
 [*.asciidoc]
 trim_trailing_whitespace = false
+
+[*.{yml,yaml}]
```
Comment (v1v) [inline .editorconfig]: ❤️

File: docker-compose-elastic-cloud.yml (hunk 1 above)
Comment (kuisathaverat) [reply docker-compose-elastic-cloud.yml]: removed form compose

File: docker-compose-elastic-cloud.yml (hunk 2 above)
Comment (kuisathaverat) [reply docker-compose-elastic-cloud.yml]: removed too

**Pull request merged**